src/monitor/Monitor.ts

```diff
@@ -104,6 +123,169 @@ export class Monitor {
     }
   }
 
+  // Report balances of all tokens on each supported chain for each relayer.
+  async reportRelayerBalances() {
+    const relayers = this.monitorConfig.monitoredRelayers;
+    for (const relayer of relayers) {
+      for (const chainId of this.monitorConfig.spokePoolChains) {
+        const tokenAddresses = this.clients.hubPoolClient.getDestinationTokensForChainId(chainId);
+
+        // Use the provider from the spoke pool to send requests to the right provider.
+        const spokePoolClient = this.clients.spokePoolClients[chainId];
+        const tokenBalances = await Promise.all(
+          tokenAddresses.map((tokenAddress) =>
+            new Contract(tokenAddress, ERC20.abi, spokePoolClient.spokePool.signer).balanceOf(relayer)
+          )
+        );
+
+        let mrkdwn = "";
+        for (let i = 0; i < tokenAddresses.length; i++) {
+          const tokenInfo = this.clients.hubPoolClient.getL1TokenInfoForL2Token(tokenAddresses[i], chainId);
+          // Convert to number of tokens for readability.
+          const balance = convertFromWei(tokenBalances[i], tokenInfo.decimals);
+          mrkdwn += `*${tokenInfo.symbol}: ${balance}*\n`;
+        }
+
+        this.logger.info({
+          at: "Monitor",
+          message: `Relayer (${relayer})'s token balances for chain ${chainId} 🚀`,
+          mrkdwn,
+        });
+      }
+    }
+  }
+
+  async reportPendingRelayerRefunds() {
+    const hubPoolClient = this.clients.hubPoolClient;
+    const latestBlockNumber = hubPoolClient.latestBlockNumber;
+    const latestBundle = hubPoolClient.getMostRecentProposedRootBundle(latestBlockNumber);
+    const chainIds = this.monitorConfig.spokePoolChains;
+
+    // Refunds that will be processed in the current bundle still pending liveness (if any).
+    let pendingFillsToRefundPerChain: FillsToRefund | undefined = undefined;
+
+    let remainingLivenessMins = 0;
+    // If latest proposed bundle has not been executed yet, we have a bundle pending liveness.
+    if (hubPoolClient.getExecutedLeavesForRootBundle(latestBundle, latestBlockNumber).length == 0) {
+      // Use the same block range as the current pending bundle to ensure we look at the right refunds.
+      const pendingBundleEvaluationBlockRanges: number[][] = [];
+      const latestBundleEvalNumbers = latestBundle.bundleEvaluationBlockNumbers;
+      for (let i = 0; i < latestBundleEvalNumbers.length; i++) {
+        const startingBlock = hubPoolClient.getNextBundleStartBlockNumber(chainIds, latestBlockNumber, chainIds[i]);
+        pendingBundleEvaluationBlockRanges.push([startingBlock, latestBundleEvalNumbers[i].toNumber()]);
+      }
+      const { fillsToRefund } = this.bundleDataClient.loadData(
```

Comment:
```suggestion
      const { fillsToRefund: pendingFillsToRefundPerChain } = this.bundleDataClient.loadData(
```

Thoughts? Would have to delete the following line of course

Reply:
This is needed because pendingFillsToRefundPerChain is declared outside the if conditional. If we define it here inside the if, it'd shadow the outside variable and would not be available to look at later outside on line 196.

Reply:
This is slightly more complex code that I hoped for but it reduces the amount of redundant code to account for 3 possible cases:

1. There's a pending bundle and a future bundle with refunds
2. There's a pending bundle but no future bundle with refunds
3. There's no pending bundle but there's a future bundle with refunds